Riot/Modules/SecretsRecovery/RecoverWithKey/SecretsRecoveryWithKeyViewController.swift

```diff
@@ -0,0 +1,279 @@
+        switch viewState {
+        case .loading:
+            self.renderLoading()
+        case .loaded:
+            self.renderLoaded()
+        case .error(let error):
+            self.render(error: error)
+        }
+    }
+    
+    private func renderLoading() {
+        self.view.endEditing(true)
+        self.activityPresenter.presentActivityIndicator(on: self.view, animated: true)
+    }
+    
+    private func renderLoaded() {
+        self.activityPresenter.removeCurrentActivityIndicator(animated: true)
+    }
+    
+    private func render(error: Error) {
+        self.activityPresenter.removeCurrentActivityIndicator(animated: true)
+
+        // TODO: Use appropriate error codes
+        if (error as NSError).domain == MXSecretStorageErrorDomain
+            && (error as NSError).code == Int(MXSecretStorageErrorCode.badMacCode.rawValue) {
+
+            self.errorPresenter.presentError(from: self,
+                                             title: VectorL10n.secretsRecoveryWithKeyInvalidRecoveryKeyTitle,
+                                             message: VectorL10n.secretsRecoveryWithKeyInvalidRecoveryKeyMessage,
+                                             animated: true,
+                                             handler: nil)
+        } else {
+            self.errorPresenter.presentError(from: self, forError: error, animated: true, handler: nil)
+        }
+    }
+    
+    private func showFileSelection() {
+        // Show only text documents
+        let documentPicker = UIDocumentPickerViewController(documentTypes: [kUTTypeText as String], in: .import)
+        documentPicker.delegate = self
+        self.present(documentPicker, animated: true, completion: nil)
+    }
+    
+    private func importRecoveryKey(from url: URL) {
+        if let recoveryKey = self.getDocumentContent(from: url) {
+            self.recoveryKeyTextField.text = recoveryKey
+            self.recoveryKeyTextFieldDidChange(self.recoveryKeyTextField)
+        } else {
+            self.errorPresenter.presentGenericError(from: self, animated: true, handler: nil)
+        }
+    }
+    
+    private func getDocumentContent(from documentURL: URL) -> String? {
+        let documentContent: String?
+        
+        do {
+            documentContent = try String(contentsOf: documentURL)
+        } catch {
+            print("Error: \(error)")
+            documentContent = nil
+        }
+        
+        return documentContent
+    }
+    
+    // MARK: - Actions
+    
+    private func cancelButtonAction() {
+        self.viewModel.process(viewAction: .cancel)
+    }
+    
+    @IBAction private func importFileButtonAction(_ sender: Any) {
+        self.showFileSelection()
+    }
+    
+    @objc private func recoveryKeyTextFieldDidChange(_ textField: UITextField) {
+        self.viewModel.recoveryKey = textField.text
+        self.updateRecoverButton()
+    }
+    
+    @IBAction private func usePassphraseButtonAction(_ sender: Any) {
```

Comment:
Using `passphrase` word is confusing in this screen
```suggestion
    @IBAction private func recoverButtonAction(_ sender: Any) {
```